libcaf_core/caf/async/file.hpp

```diff
@@ -52,7 +56,10 @@ class file_reader {
   template <class Step, class... Steps>
   void pull(size_t n, Step& step, Steps&... steps) {
     if (file_ == nullptr) {
-      file_ = fopen(path_.c_str(), "r");
+      if constexpr (std::is_same_v<T, std::byte>)
+        file_ = fopen(path_.c_str(), "rb");
+      else
+        file_ = fopen(path_.c_str(), "r");
```

Comment:
```suggestion
      file_ = fopen(path_.c_str(), mode());
```

With `mode` being a private member function, e.g.:

```c++
  static constexpr const char* mode() noexcept {
      if constexpr (std::is_same_v<T, std::byte>)
        return "rb";
      else
        return "r";
  }
```